src/math/window.c

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+
+#include <sof/audio/format.h>
+#include <sof/math/decibels.h>
+#include <sof/math/log.h>
+#include <sof/math/trig.h>
+#include <sof/math/window.h>
+#include <errno.h>
+#include <stdint.h>
+
+#define WIN_ONE_Q15 INT16_MAX
+#define WIN_ONE_Q31 INT32_MAX
+#define WIN_05_Q31 Q_CONVERT_FLOAT(0.5, 31)
+
+#define WIN_TWO_PI_Q28 Q_CONVERT_FLOAT(6.2831853072, 28)
+
+#define WIN_085_Q31 Q_CONVERT_FLOAT(0.85, 31)
+#define WIN_LOG_2POW31_Q26 Q_CONVERT_FLOAT(21.4875625974, 26)
+
+/* Exact
+ * #define WIN_HAMMING_A0_Q30 Q_CONVERT_FLOAT(25.0 / 46.0, 30)
+ * #define WIN_HAMMING_A1_Q30 Q_CONVERT_FLOAT(1 - 25.0 / 46.0, 30)
+ */
+
+/* Common approximations to match e.g. Octave */
+#define WIN_HAMMING_A0_Q30 Q_CONVERT_FLOAT(0.54, 30)
+#define WIN_HAMMING_A1_Q30 Q_CONVERT_FLOAT(0.46, 30)
+
+/**
+ * \brief Return matrix of Q1.15 ones
+ * \param[in,out] x[][] Output two dimensional array
+ * \param[in] rows Number of rows
+ * \param[in] columns Number of columns
+ *
+ */
+
+/* TODO: Decide how to present one and two dimensional arrays, for matrix
+ * operations library. pointer arithmetic like below vs. x[][]
+ */
+static void mat_ones_16b(int16_t *x, int rows, int columns)
+{
+	int i;
+	int j;
+
+	for (i = 0; i < rows; i++) {
+		for (j = 0; j < columns; j++)
+			*(x + columns * i + j) = WIN_ONE_Q15;
```

Comment:
`x[columns * i + j]` looks simpler to me

Reply:
Yep. I've changed the matrix definition, it's now a struct to hold some needed side information. Since this is not compatible so I will just do a "memset" to initialize it for window.